plugins/blockvolume/blockprovider/gluster-block/glusterblock.go

```diff
@@ -0,0 +1,272 @@
+package glusterblock
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/pkg/size"
+	"github.com/gluster/glusterd2/plugins/blockvolume/blockprovider"
+	"github.com/gluster/glusterd2/plugins/blockvolume/utils"
+
+	"github.com/gluster/gluster-block-restapi/client"
+	"github.com/gluster/gluster-block-restapi/pkg/api"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+const providerName = "gluster-block"
+
+func init() {
+	log.WithField("name", providerName).Infof("Registering block provider")
+	blockprovider.RegisterBlockProvider(providerName, newGlusterBlock)
+}
+
+// GlusterBlock implements block Provider interface. It represents a gluster-block
+type GlusterBlock struct {
+	client     client.GlusterBlockClient
+	ClientConf *ClientConfig
+}
+
+func newGlusterBlock() (blockprovider.Provider, error) {
+	var (
+		gb         = &GlusterBlock{}
+		clientConf = &ClientConfig{}
+		opts       = []client.OptFuncs{}
+	)
+
+	clientConf.ApplyFromConfig(viper.GetViper())
+	gb.ClientConf = clientConf
+
+	opts = append(opts,
+		client.WithAuth(clientConf.User, clientConf.Secret),
+		client.WithTLSConfig(&client.TLSOptions{CaCertFile: clientConf.CaCertFile, InsecureSkipVerify: clientConf.Insecure}),
+	)
+
+	gbClient, err := client.NewClientWithOpts(clientConf.HostAddress, opts...)
+	if err != nil {
+		return nil, err
+	}
+	gb.client = gbClient
+
+	return gb, nil
+}
+
+// CreateBlockVolume will create a gluster block volume with given name and size having `hostVolume` as hosting volume
+func (g *GlusterBlock) CreateBlockVolume(name string, size int64, hosts []string, hostVolume string, options ...blockprovider.BlockVolOption) (blockprovider.BlockVolume, error) {
```

Comment:
if you have more than 5 input parameters, why can't we have a structure pointer as an input?